RemoteImageBuffer uses incorrect connection when validating messages

kkinnunen-apple opened the pull request:
#### 14b727ba8c4a48c47ff6d2bfafd5ed3ff3606e98
<pre>
RemoteImageBuffer uses incorrect connection when validating messages
<a href="https://bugs.webkit.org/show_bug.cgi?id=297850">https://bugs.webkit.org/show_bug.cgi?id=297850</a>
<a href="https://rdar.apple.com/159089435">rdar://159089435</a>

Reviewed by Simon Fraser.

Notify the RemoteRenderingBacknend connection, the messages
come from this.

* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp:

Canonical link: <a href="https://commits.webkit.org/299149@main">https://commits.webkit.org/299149@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63726304f18a505d7e5102d38401950a2a2ed699

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117900 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37578 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28210 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124044 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69933 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a36dc542-20d0-49ca-b85e-5748c52f4649) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38270 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46160 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89466 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59084 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/79a6897b-1189-473a-b5a2-ce8509853c10) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120852 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30471 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105720 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69962 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c9935bb2-c13e-46d8-b211-c9e4fcecf0a6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29535 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67708 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99891 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24015 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127127 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44803 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98136 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45164 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101945 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97924 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24934 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43323 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21300 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41232 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44675 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50349 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44135 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47480 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45824 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (approved):
"RemoteRenderingBacknend" in commit message.

kkinnunen-apple committed:
RemoteImageBuffer uses incorrect connection when validating messages
https://bugs.webkit.org/show_bug.cgi?id=297850
rdar://159089435

Reviewed by Simon Fraser.

Notify the RemoteRenderingBacknend connection, the messages
come from this.

* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp:

Canonical link: https://commits.webkit.org/299149@main
